packages/@angular/cli/tasks/serve.ts

```diff
@@ -108,7 +108,7 @@ export default Task.extend({
       stats: statsConfig,
       inline: true,
       proxy: proxyConfig,
-      compress: serveTaskOptions.target === 'production',
+      compress: serveTaskOptions.target === 'production' && serveTaskOptions.compress,
```

Comment:
This should just be `serveTaskOptions.compress` once the option default is based on build target.